.github/workflows/gating.yaml

```diff
@@ -101,3 +113,57 @@ jobs:
         ignore: DL3041
         failure-threshold: warning
 
+  build-image:
+    name: Build Cachi2 image and run smoketest on it
+    runs-on: ubuntu-latest
+    steps:
+      - name: add checkout action...
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Build Cachi2 image
+        uses: redhat-actions/buildah-build@v2
+        with:
+          image: ${{ env.IMAGE_NAME }}
+          tags: ${{ github.sha }}
+          arch: amd64
+          containerfiles: |
+            ./Containerfile
+
+      - name: Check image created and Cachi2 version
+        run: |
+          buildah images | grep '${{ env.IMAGE_NAME }}'
+          podman run -t ${{ env.IMAGE_NAME }}:${{ github.sha }} --version
+
+      - name: Run smoketest on built images
+        run: |
+          git clone -b ${{ env.TEST_BRANCH }} ${{ env.TEST_REPO }} ${{ env.TEST_FOLDER }}
+          podman run -t -v ${{ env.TEST_FOLDER }}:${{ env.TEST_FOLDER }} \
+            ${{ env.IMAGE_NAME }}:${{ github.sha }} \
+            fetch-deps \
+            --source ${{ env.TEST_FOLDER }} \
+            --output ./cachi2-output \
+            --package '{"path": ".", "type": "gomod"}'
+
+     # Authenticate to container image registry to push the image
+      - name: Podman Login
+        uses: redhat-actions/podman-login@v1
+        if: github.ref == 'refs/heads/main'
+        with:
+          registry: ${{ env.IMAGE_REGISTRY }}
+          username: ${{ secrets.QUAY_USER }}
+          password: ${{ secrets.QUAY_PWD }}
+
+      - name: Push image to Quay.io registry
+        if: github.ref == 'refs/heads/main'
```

Comment:
I think this won't run for a [release event](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release). We could probably add something like "or event == release", but maybe we could come back to release workflows later.

Maybe releasing should just be taking `quay.io/containerbuildsystem/cachi2:${{ github.sha }}` and adding the current tag to it, maybe we don't need to build a whole new image

Reply:
Btw. @brunoapimentel do you have ideas how versioning of the cachi2 image will be done in HACBS? Should we have tags for each released semantic version? Are just the commit-hash tags good enough? https://github.com/redhat-appstudio/build-definitions/pull/218#discussion_r1011502124

Reply:
The commit hash would be enough for HACBS, but we could also use semantic versions. In the latter case, I would also add the image sha, this way we'd make sure we're pinning a specific image version to the pipeline.

The image versions are being bumped manually in HACBS for now. I have filed a ticket so we can decide on a process on how to update the image tags there.

Reply:
Image sha as in manifest digest? Pinning to digest is the universal way to pin a specific image
```
quay.io/containerbuildsystem/cachi2@sha256:bc87b6bd5549b4b4cb5d459a5edaa0e705e2ac74d0ae7fcdb029cac3da82be76
```

Or image sha as in commit sha? I think we should push commit shas on merge to main and tag those shas with the semver tag on release

Reply:
+1 for adding also image sha, do you want to handle releases here? or in a separate PR?

Reply:
I'd vote for a separate PR with a tiny design first

Reply:
I dropped part for releases for now and filed ticket CLOUDBLD-12459 for design and investigation. Any more comments for this PR?